Waiting for review.

unbreaking-V opened the pull request.

unbreaking-V committed:
Waiting for review.

unbreaking-V committed:
[#3] Rebuild serwer

Makefile: the makefile has been redesigned, and now two files are compiled, one client and one server.
client.cpp: Modified the client, the client now sends a message to the server, using the command `send`.
server.cpp: 1.Server handles several connections at the same time

            2.Server handles all connections

            3.Server can close a connection with a client that did not receive a response

            4.Server was built using OOP

server.hpp: The header file contains all the libraries, functions and variables needed for the server

ivysochyn commented:
Reword your commits to display changes they provide. Use the next naming scheme:
`[#<issue number>] <file/group was changed>: <file/group was changed>: Commit message`

For example:
`[#1] README.md: Fix typo`

In order to reword your commits use `git rebase`, inside editor mark corresponding commits with `r` instead of `pick`. In order to run rebase on your branch use:
`git rebase -i <commit hash>`

ivysochyn commented on Makefile:
```diff
@@ -1,8 +1,14 @@
+CC = g++  
+CFLAGS_OBJ =  -Wall -Wextra -pedantic -g 
+
+normal: $(TARGET)
```
`normal` should be renamed to `main`.

ivysochyn commented on main.cpp:
```diff
@@ -0,0 +1,17 @@
+int main(int argc, char **argv)
+{
+    server.init();	
+	//actual main loop
```
Replace all the tabs with four spaces, we should follow codestyle.

ivysochyn commented on client.cpp:
```diff
@@ -1,21 +1,17 @@
-#include <string.h>
-#include <unistd.h>
-#include <stdio.h>
+#include "server.hpp"
```
Why are we using `server.hpp` to include? I believe we should have `client.cpp` independent of server.

ivysochyn replied:
Too many spaces as well, follow the code style.

ivysochyn commented on main.cpp:
```diff
@@ -0,0 +1,17 @@
+#include <cstdint>
+#include <vector>
+#include <string>
+
```
Is this new line is needed here?

unbreaking-V replied:
Remove include

ivysochyn commented on server.cpp:
```diff
@@ -1,28 +1,189 @@
+       exit(EXIT_FAILURE);
+    }
+        
+    //Clear an masterfds and tempfds
```
Use space after comment slashes: `// Text`

ivysochyn commented on server.cpp:
```diff
@@ -1,28 +1,189 @@
+    con_info.sin_addr.s_addr = htons(INADDR_ANY);
+    //zero the input buffer before use to avoid random data appearing in first receives
+    bzero(input_buffer,INPUT_BUFFER_SIZE);
+
```
Inconsistent use of new line characters.

ivysochyn commented on server.cpp:
```diff
@@ -1,28 +1,189 @@
+
+void Server::initializeSocket()
+{   
+    /*setsockopt - sets flags on the socket
```
space

ivysochyn commented on server.cpp:
```diff
@@ -1,28 +1,189 @@
+void Server::startListen()
+{
+    /*Prepare a socket for peeling requests 
+     * and allows you to determine the size(3) 
```
Grammar errors

ivysochyn commented on server.cpp:
```diff
@@ -1,28 +1,189 @@
+
+void Server::handleNewConnection()
+{   
+    socklen_t addrlen = sizeof (client_addr);
```
Space between `sizeof` and `argument`.

ivysochyn commented on server.cpp:
```diff
@@ -1,28 +1,189 @@
+{   
+    int nbytesrecv = recv(fd, input_buffer, INPUT_BUFFER_SIZE, 0);
+    
+    printf("%s\n", input_buffer);
```
Do we need to have this `printf` here?

unbreaking-V replied:
Removing printf

ivysochyn commented on server.cpp:
```diff
@@ -1,28 +1,189 @@
+    {
+        /*If the client does not send any messages,
+         * it means that it has finished its work and left,
+         * and there is no need to maintain a connection with it
```
Grammar

unbreaking-V replied:
Fixed

ivysochyn commented on server.hpp:
```diff
@@ -0,0 +1,66 @@
+#include<cstdio>
+#include<cstdlib>
+#include<cstring>
+#include<fcntl.h>
```
Do we really use all those headers?

unbreaking-V replied:
Reviews include

ivysochyn commented on server.hpp:
```diff
@@ -0,0 +1,66 @@
+        //input buffer 
+        char input_buffer[INPUT_BUFFER_SIZE];
+
+        void (*newConnectionCallback) (uint16_t fd);
```
We're not using those callbacks, do we need them defined?

unbreaking-V replied:
Removing callback function

ivysochyn commented on server.hpp:
```diff
@@ -0,0 +1,66 @@
+        void initializeSocket();
+        void bindSocket();
+        void startListen();
+        void shutdown();
```
I don't implementation of shutdown, do we need it?

unbreaking-V replied:
Remove shutdown function

ivysochyn reviewed (changes requested):
Follow code style with spaces/tabs, comments and brackets.
Review used headers to ensure we are not including not needed stuff.
Review not used/implemented functions.

unbreaking-V committed:
[#3] Review and update code style

main.cpp: removing unnecessary inludes
server.cpp: updating the code style, removing unused functions and replacing tabs with spaces
server.hpp: removing unnecessary inludes and functions
